### PR TITLE
Protocol tests json

### PR DESCRIPTION
**Pull request by arvidn**
the main motivation for this patch is to add regression tests for the JSON dictionary representations for all network protocols, to ensure they don't accidentally change to become incompatible.

We already have this test for the streamable binary representation of messages (`test_network_protocol_files.py`).

In order to create this test, I extended the visitor in `build_network_protocol_files.py` to not only pass the network message object, but also the variable name. this allows that script to not just generate the test data, but also the test itself. This makes it much more convenient (and likely) to add new protocol messages to this test when we add them to the protocol.

With this new capability, I also made the existing test be generated by the build script.

### CODE REVIEW

File: tests/util/build_network_protocol_files.py [hunk 1]
```diff
@@ -155,13 +155,106 @@ def visitor(obj: Any) -> None:
...
+
+    visit_all_messages(visitor)
+
+    return result
+
+
 if __name__ == "__main__":
-    filename = get_network_protocol_filename()
-    data = get_protocol_bytes()
-    if os.path.exists(filename):
```
Comment (arvidn) [inline tests/util/build_network_protocol_files.py:161]: opening the file with `"w"` already truncates any existing file

File: tests/util/test_network_protocol_files.py [hunk 2]
```diff
@@ -29,670 +17,461 @@ def parse_blob(input_bytes):
...
-    ret: List[str] = []
-    mod_name = mod.__name__
-    for sym in dir(mod):
-        obj = getattr(mod, sym)
-        if hasattr(obj, "__module__") and obj.__module__ == mod_name:
-            ret.append(sym)
-    return set(ret)
-
-
-def test_missing_messages_state_machine() -> None:
```
Comment (arvidn) [inline tests/util/test_network_protocol_files.py:535]: I moved these tests into a separate file, since they are not generated

**Review by fchirica — DISMISSED** [review]
LGTM

**Review by fchirica — APPROVED** [review]

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**